### PR TITLE
Improve the Post_Collection plucking

### PR DESCRIPTION
**Pull request by lucatume**
This PR adds the following methods to the `Tribe__Utils__Post_Collection` class:

* `pluck` to pluck a post field, taxonomy or custom field from the post collection.
* `pluck_taxonomy` to pluck one or more taxonomy terms from each post in the collection.
* `pluck_field` to pluck a post field (e.g. the `post_title`) from all posts in the collection.

The collection is used to create sets of results from the `Tribe__Repository::collect` method:

```php
// List all titles of the found posts.
$titles = tribe_events()->where('starts_after', 'today')->collect()->pluck('post_title');

// The same could be written using the `pluck_field` method to skip the field type detection.
$parents = tribe_events()->where('starts_after', 'today')->collect()->pluck_field('post_parent');

// Get the names of the tags assigned to each post in the collection.
$single = false;
$tags = tribe_events()->where('starts_after', 'today')->collect()->pluck('post_tag', $single);

// The same could be written using the `pluck_taxonomy` method to skip the field type detection.
$tags = tribe_events()->where('starts_after', 'today')->collect()->pluck_taxonomy('post_tag', $single);

// Get the UTC start date of each event in the collection.
$utc = tribe_events()->where('starts_after', 'now')->collect()->pluck('_EventStartDateUTC');

// Same as...
$utc = tribe_events()->where('starts_after', 'now')->collect()->pluck_meta('_EventStartDateUTC');

// Get a list of Organizer IDs for each event in the collection.
$single = false;
$orgs = tribe_events()->where('starts_after', 'now')->collect()->pluck('_EventOrganizerID', $single);

// Same as...
$orgs = tribe_events()->where('starts_after', 'now')->collect()->pluck_meta('_EventOrganizerID', $single);
```

See more in the tests.

### CODE REVIEW

**Comment by lucatume** [comment]
I've iterated on this a bit and added the `pluck_combine` method, think GraphQL-ish.

```php
$event_dates_and_timets = tribe_events()

```

File: src/Tribe/Traits/With_Post_Attribute_Detection.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	 *
+	 * @since 4.7.19 Created in the `Tribe__Repository` class.
+	 * @since TBD Refactored out of the `Tribe__Repository` class into this trait.
+	 *
+	 * @param string $key The field to check.
+	 *
+	 * @return bool Whether the key indicates a taxonomy of the post type or not.
+	 */
+	protected function is_a_taxonomy( $key ) {
+		if(!isset($this->taxonomies))	{
```
Comment (jesseeproductions) [inline src/Tribe/Traits/With_Post_Attribute_Detection.php:70]: Please check the formatting here.

File: src/Tribe/Utils/Post_Collection.php [hunk 2]
```diff
@@ -24,6 +36,216 @@ public function __construct( array $items ) {
...
+	 * @param string|array<string>|array<array> $value_fields Either a single field name to populate the values with;
+	 *                                                        a list of fields, each plucked with default settings;
+	 *                                                        a map of fields to fetch, each defining a `single` and
+	 *                                                        `args` key to define the pluck `$single` and `$args`
+	 *                                                        parameters where applicable.
+	 *                                                        Additionally an `as` parameter can be specifed to alias
+	 *                                                        the field in the results.
+	 *
+	 * @return array<int|string,string|array> A list of plucked fields or a map of plucked fields keyed by the
+	 *                                        specified
```
Comment (jesseeproductions) [inline src/Tribe/Utils/Post_Collection.php:62]: The comments look fine when directly looking at the file, but here specified field. could be on one line.

**Comment by borkweb** [comment]
Wow...this is a pretty darn great addition, @lucatume.

**Pull request merged**